Sources/Core/Microsoft.StreamProcessing/Operators/EquiJoin/FixedInterval/FixedIntervalEquiJoinPipe.cs

```diff
@@ -12,6 +12,8 @@
 namespace Microsoft.StreamProcessing
 {
     [DataContract]
+    [KnownType(typeof(EndPointHeap))]
+    [KnownType(typeof(EndPointQueue))]
```

Comment:
We should investigate to see if we can put these KnownType annotations on the interface or somewhere else instead of LITERALLY EVERY SINGLE CLASS

Reply:
I agree. As someone who is not familiar with KnownType it was also a pain to debug!

---
In reply to: [320965069](https://github.com/microsoft/Trill/pull/108#discussion_r320965069) [](ancestors = 320965069)